src/asm/fstack.c

```diff
@@ -219,16 +219,16 @@ bool yywrap(void)
 			contextStack->fileInfo = (struct FileStackNode *)fileInfo;
 		}
 
-		fileInfo->iters[0]++;
 		/* If this is a FOR, update the symbol value */
-		if (contextStack->forName) {
+		if (contextStack->forName && fileInfo->iters[0] <= contextStack->nbReptIters) {
 			contextStack->forValue += contextStack->forStep;
 			struct Symbol *sym = sym_AddSet(contextStack->forName,
 				contextStack->forValue);
 
 			if (sym->type != SYM_SET)
 				fatalerror("Failed to update FOR symbol value\n");
 		}
+		fileInfo->iters[0]++;
```

Comment:
I would add a comment noting that this is there in order to report the iteration that just finished, should `sym_AddSet` fail.